### ISSUE TITLE
Filtering adapters by card-location returns KeyError: 'card-location'

### ISSUE THREAD

**Issue by leopoldjuergen** [zhmcclient/python-zhmcclient#271]
### Actual behavior
```
#!/usr/bin/env python

import sys
import zhmcclient
import requests.packages.urllib3
requests.packages.urllib3.disable_warnings()

# Set these variables for your environment:
hmc_host = "xxx.xxx.xxx"
hmc_userid = "xxxx"
hmc_password = "xxxxx"

session = zhmcclient.Session(hmc_host, hmc_userid, hmc_password)
client = zhmcclient.Client(session)
cpcs = client.cpcs.list()
if len(cpcs) != 1:
    print("Could not find CPC in status active")
    sys.exit(1)
cpc = cpcs[0]

print("Found CPC %s " % (cpc.name))

filter_args = {
#    'adapter-family': 'osa',
    'card-location': 'A13B-LG36J.*',
#    'state': ['stand-by', 'reserved', 'unknown'],
}

osa_adapters = cpc.adapters.list(filter_args=filter_args)
for osa_adapter in osa_adapters:
    osa_adapter.pull_full_properties()
    print(osa_adapter.properties)
```
I use the above script to filter adapters. If I use 'card-location' **without** any other filter I got following exception:
```
Traceback (most recent call last):
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_resource.py", line 247, in get_property
    return self._properties[name]
KeyError: 'card-location'

During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "./adapter_filter.py", line 43, in <module>
    filter_args=filter_args)
  File "<decorator-gen-61>", line 2, in list
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_logging.py", line 202, in log_api_call
    result = func(*args, **kwargs)
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_adapter.py", line 177, in list
    if self._matches_filters(resource_obj, client_filters):
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_manager.py", line 343, in _matches_filters
    if not self._matches_prop(obj, prop_name, prop_match):
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_manager.py", line 354, in _matches_prop
    prop_value = obj.get_property(prop_name)
  File "<decorator-gen-6>", line 2, in get_property
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_logging.py", line 202, in log_api_call
    result = func(*args, **kwargs)
  File "/home/vagrant/venv/local/lib/python3.4/site-packages/zhmcclient/_resource.py", line 252, in get_property
    return self._properties[name]
KeyError: 'card-location'
```
Using 'adapter-family' with 'card-location' work fine. 

### Execution environment

* zhmcclient version: version 0.12.1.dev25
* Operating system (type+version): Ubuntu 14.04
* HMC version:
* CPC version:

**Comment by andy-maier** [comment]
What I can say so far, is that `card-location` is a property that is filtered on the client side. More investigation needed.

**Comment by andy-maier** [comment]
This was caused by the fact that not all adapters have the `card-location` property. For example, Hipersockets adapters do not have it. This was fixed in PR #272 by tolerating resources that don't have a specified filter property and by treating such resources as non-matching.
